crates/rmf_site_picking/src/select/events.rs

```diff
@@ -23,15 +23,25 @@ pub struct Hover(pub Option<Entity>);
 
 /// Used as an event to command a change in the selected entity.
 #[derive(Default, Debug, Clone, Copy, Deref, DerefMut, Event)]
-pub struct Select(pub Option<SelectionCandidate>);
+pub struct Select {
+    #[deref]
+    pub selection_candidate: Option<SelectionCandidate>,
+    pub multi_select: bool,
+}
 
 impl Select {
-    pub fn new(candidate: Option<Entity>) -> Select {
-        Select(candidate.map(|c| SelectionCandidate::new(c)))
+    pub fn new(candidate: Option<Entity>, multi_select: bool) -> Select {
+        Select {
+            selection_candidate: candidate.map(|c| SelectionCandidate::new(c)),
+            multi_select,
+        }
     }
 
-    pub fn provisional(candidate: Entity) -> Select {
-        Select(Some(SelectionCandidate::provisional(candidate)))
+    pub fn provisional(candidate: Entity, multi_select: bool) -> Select {
```

Comment:
Same as before, let's take the `multi_select` argument out of here and just use the `.multi_select(_)` builder pattern.